Cargo.toml

```diff
@@ -15,3 +15,4 @@ log = "0.4.17"
 num = "0.4.0"
 itertools = "0.10.2"
 serde_json = "1.0.79"
+push-trait = "0.6.0"
```

Comment:
We need to be able to specify on type that it has `push` behaviour defined. I found [this crate](https://docs.rs/push-trait/latest/push_trait/index.html) but it might be good idea to create our own, so that we will be able to tailor it to our needs. The one I found is not even hosted on `GitHub` (at least I haven't managed to find its gh page) so we can't even patch it if needed...